FV3/atmos_cubed_sphere/tools/coarse_grained_diagnostics.F90

```diff
@@ -637,6 +672,18 @@ subroutine maybe_allocate_reference_array(Atm, coarse_diagnostic)
              allocate(Atm(tile_count)%lagrangian_tendency_of_hydrostatic_pressure(isd:ied,jsd:jed,1:npz))
              Atm(tile_count)%lagrangian_tendency_of_hydrostatic_pressure = 0.0
           endif
+       elseif (ends_with(coarse_diagnostic%name, 'fv_sat_adj_t_dt_coarse')) then
+          if (.not. allocated(Atm(tile_count)%fv_sat_adj_tendency_diag%fv_sat_adj_t_dt)) then
+              allocate(Atm(tile_count)%fv_sat_adj_tendency_diag%fv_sat_adj_t_dt(is:ie,js:je,1:npz))
+              Atm(tile_count)%fv_sat_adj_tendency_diag%fv_sat_adj_t_dt(is:ie,js:je,1:npz) = 0.0
+          endif
+          coarse_diagnostic%data%var3 => Atm(tile_count)%fv_sat_adj_tendency_diag%fv_sat_adj_t_dt(is:ie,js:je,1:npz)
+       elseif (ends_with(coarse_diagnostic%name, 'fv_sat_adj_qv_dt_coarse')) then
```

Comment:
So the column-integrated versions always get allocated but the 3D ones maybe do?

Reply:
These deferred allocation statements are needed for allocating the full-resolution field, which is needed either if we are computing the 3D coarse field or the 2D coarse vertically integrated field.  Since I am checking that the name "ends with" `fv_sat_adj_qv_dt_coarse`, and only allocate memory if it was not allocated already, this works in both cases.